default_dial_option_server_option_test.go

```diff
@@ -74,6 +75,29 @@ func (s) TestDisableGlobalOptions(t *testing.T) {
 	internal.ClearGlobalDialOptions()
 }
 
+type testLateApplyDialOption struct{}
+
+func (do *testLateApplyDialOption) DialOption(parsedTarget url.URL) DialOption {
+	if parsedTarget.Scheme == "passthrough" {
+		return WithTransportCredentials(insecure.NewCredentials()) // credentials provided, should pass NewClient.
+	}
+	return WithReadBufferSize(10) // no credentials, should fail NewClient
+}
+
+func (s) TestGlobalLateApplyDialOption(t *testing.T) {
```

Comment:
It took a good bit of learning this to figure out what it's even doing.  A short comment to say "configure a global late dial option that produces TransportCredentials for channels using the passthrough scheme, which makes those channels valid but others invalid" would be really helpful.

Reply:
Added comment: "TestGlobalLateApplyDialOption configures a global dial option that produces transport credentials for channels using "passthrough" scheme. Channels that use the passthrough scheme should this be successfully created due to picking up transport credentials, whereas other channels should fail at creation due to not having transport credentials."